### DISCUSSION TITLE
[@types/discord-rpc] tries to import nonexistent `listener` from `'events'`

### DISCUSSION THREAD

**Discussion by Nowaaru** [DefinitelyTyped/DefinitelyTyped#59650] [Issues with a @types package]
Package `discord-rpc` attempts to import the `Listener` interface from `node/events` but it does not exist:

```log
✖ 16 problems (0 errors, 16 warnings)

node_modules/@types/discord-rpc/index.d.ts:11:24 - error TS2305: Module '"events"' has no exported member 'Listener'.

11 import { EventEmitter, Listener } from 'events';
                          ~~~~~~~~

Found 1 error.

Error: Process completed with exit code 2.
```

Node version 16.14.2 LTS is installed.

[a comment by an automated account is not shown]

**Comment by HanchaiN** [comment]
I suppose that it should import the `Listener` interface from [`events`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/events) instead.

But the following piece of code from `events` and the similar one from [`node/events`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts) can be used to fix the issue.
```export type Listener = (...args: any[]) => void;```

Noted that they use `listener: (...args: any[]) => void` without type declaration in `node/events`.

**Reply by eddieqiii** [reply]
Looks like the [`events` package](https://www.npmjs.com/package/events) you're referring to isn't the most common use case, and is outdated:

> This implements the Node.js events module for environments that do not have it, like browsers.

> events currently matches the Node.js 11.13.0 API.

> You usually do not have to install events yourself! If your code runs in Node.js, events is built in. If your code runs in the browser, bundlers like browserify or webpack also include the events module.

I think the relevant type definitions are [the ones in `@types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/events.d.ts), which this change would break with.

(Mentioning #59187 so this discussion shows up there)
